duckduckgo/iOS
# Feedback disambiguation

Task/Issue URL: https://app.asana.com/0/856498667320406/1109166030178657
Tech Design URL:
CC:

**Description**:
Updated feedback flow with categories and subcategories to disambiguate the feedback.

Q: When are Pixels fired?
A: Pixels are fired for each "submit feedback" action, which is:
 - For positive flow: 'rate app', 'no thanks' (on positive feedback screen) and 'submit feedback' (on feedback form).
 - For positive flow: Every time user presses 'submit feedback'.

Q: Where we can see feedback?
A: Asana [Mobile/AppFeedback](https://app.asana.com/0/1114857721287938/list) project and [Grafana](https://grafana.duckduckgo.com/d/YDs3FwCmz/ios-app-feedback)

Q: What about old feedback form?
A: We leave it as it is for now.



**Steps to test this PR**:
UI-related corner cases:
1. iPhone SE in landscape mode.
2. iPad of regular size.
3. iPad Pro 12,9 inch.

For all of the above, observer if keyboard&scrolling works properly on Feedback Form (in both regular and Negative Feedback->Broken website flow).

Feedback Flow - Category & Subcategory
1. Navigate to settings -> send feedback.
2. Select negative flow.
3. Select feedback category other than 'None of these' and 'Website issues'.
4. Select feedback subcategory other than 'None of these'
5. Ensure feedback form is populated with respective choices.

Feedback Flow - Category & Other
1. Navigate to settings -> send feedback.
2. Select negative flow.
3. Select feedback category other than 'None of these' and 'Website issues'.
4. Select feedback subcategory 'None of these'
5. Ensure feedback form is populated with respective choices (subtitle should be of generic type).

Feedback Flow - Website issues
1. Navigate to settings -> send feedback.
2. Select negative flow.
3. Select feedback category 'Website issues'.
4. Ensure feedback form has additional URL field and labels are related to Website issues.

Feedback Flow - Other issues
1. Navigate to settings -> send feedback.
2. Select negative flow.
3. Select feedback category 'None of these'
4. Ensure feedback form has generic title/subtitle.

Feedback Flow - Positive flow -> Rate app
1. Navigate to settings -> send feedback.
2. Select positive flow.
3. Tap 'rate the app' - AppStore should be opened.

Feedback Flow - Positive flow -> Share Details
1. Navigate to settings -> send feedback.
2. Select positive flow.
3. Tap 'Share Details' - Feedback form should be presented with correct image and title/subtitle/placeholder.

Feedback Flow - Positive flow -> No thanks
1. Navigate to settings -> send feedback.
2. Select positive flow.
3. Tap 'No thanks' - view is dismissed.

Report broken site
1. Navigate to some webpage.
2. Options -> Report broken site
3. See that this form works as expected.


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
